MiniTravelBlog/MiniTravelBlog/MiniPost.swift

```diff
@@ -0,0 +1,92 @@
+//
+//  MiniPost.swift
+//  MiniTravelBlog
+//
+//  Created by Charlotte Liang on 3/23/22.
+//
+
+import UIKit
+import FirebaseAuth
+import FirebaseCore
+import FirebaseStorage
+import FirebaseFirestore
+
+struct MiniPost {
+  static func getPostURL(imageName: String, completion: @escaping (URL, String) -> Void) {
+    if FirebaseApp.app() == nil {
+      FirebaseApp.configure()
+    }
+
+    let ref = Storage.storage().reference().child(imageName)
+    ref.downloadURL { url, error in
+      if let error = error {
+        print("Error fetching URL: \(error)")
+        completion(url!, "")
+        return
+      } else {
+        let db = Firestore.firestore()
+        db.collection("Posts").document("post")
+          .getDocument { document, error in
+            if let document = document, document.exists {
+              guard let data = document.data() else {
+                print("Document data was empty.")
+                completion(url!, "")
+                return
+              }
+              print("Current data: \(data)")
+              let description = data["description"] as? String
+              completion(url!, description ?? "")
+            } else {
+              print("Document does not exist")
+              completion(url!, "test")
+            }
+          }
+      }
+    }
+    
+  }
+
+  static func getPost(imageName: String, completion: @escaping (UIImage, String) -> Void) {
+    if FirebaseApp.app() == nil {
+      FirebaseApp.configure()
+    }
+
+    let ref = Storage.storage().reference().child(imageName)
+    ref.getData(maxSize: 20 * 1024 * 2048) { (data: Data?, error: Error?) in
```

Comment:
If you use `FirebaseStorageSwift` you can use the async version: https://github.com/firebase/firebase-ios-sdk/blob/90b117f6f9f40ef5b627ffa3769105c01456cbce/FirebaseStorageSwift/Sources/AsyncAwait.swift#L29

Reply:
Done.